### PR TITLE
Refresh metadata when address book changes for confirm transaction and personal sign views

### PR DESCRIPTION
**Pull request by KillariDev**
partial fix for https://github.com/DarkFlorist/TheInterceptor/issues/105

Doesn't work with approve access dialog yet

### CODE REVIEW

File: extension/app/ts/background/popupMessageHandlers.ts [hunk 1]
```diff
@@ -169,6 +169,18 @@ export async function refreshSimulation(ethereumClientService: EthereumClientSer
...
 
+export async function refreshPopupConfirmTransactionMetadata(simulator: Simulator, userAddressBook: UserAddressBook, { data }: RefreshConfirmTransactionMetadata) {
+	console.log('refreshPopupConfirmTransactionMetadata')
+	const addressMetadata = await getAddressBookEntriesForVisualiser(simulator, data.visualizerResults.map((x) => x.visualizerResults), data.simulationState, userAddressBook)
+	return await sendPopupMessageToOpenWindows({
+		method: 'popup_confirm_transaction_simulation_state_changed' as const,
+		data: {
+			...data,
+			addressBookEntries: addressMetadata,
+		}
```
Comment (MicahZoltu) [inline extension/app/ts/background/popupMessageHandlers.ts:180]: Style Suggestion: Consider one line here, doesn't feel like we are improving readability in this case.  For consistency, consider this style when all you are doing is a copy plus one addition/change as done here.
```suggestion
		data: { ...data, addressBookEntries: addressMetadata }
```

File: extension/app/ts/background/windows/personalSign.ts [hunk 2]
```diff
@@ -43,6 +43,81 @@ function reject() {
...
+			method: 'popup_personal_sign_request',
+			data: {
+				activeAddress,
+				type: 'NotParsed' as const,
+				simulationMode: simulationMode,
+				requestId: requestId,
+				message: params.params[0],
+				account: getAddressMetaData(params.params[1], userAddressBook),
+				method: params.method,
+				params: params
```
Comment (MicahZoltu) [inline extension/app/ts/background/windows/personalSign.ts:58]: If you have a local parameter with the same name as the property, you can just list the local without the property name.  In fact, you already do this for `activeAddress`.
```suggestion
				activeAddress,
				type: 'NotParsed' as const,
				simulationMode,
				requestId,
				message: params.params[0],
				account: getAddressMetaData(params.params[1], userAddressBook),
				method: params.method,
				params,
```

File: extension/app/ts/background/windows/personalSign.ts [hunk 3]
```diff
@@ -43,6 +43,81 @@ function reject() {
...
+				requestId: requestId,
+				message: parsed,
+				account: getAddressMetaData(params.params[0], userAddressBook),
+				method: params.method,
+				addressBookEntries: {
+					owner: getAddressMetaData(parsed.message.owner, userAddressBook),
+					spender: getAddressMetaData(parsed.message.spender, userAddressBook),
+					verifyingContract: getAddressMetaData(parsed.domain.verifyingContract, userAddressBook)
+				},
+				params: params
```
Comment (MicahZoltu) [inline extension/app/ts/background/windows/personalSign.ts:80]:
```suggestion
				activeAddress,
				type: 'Permit' as const,
				simulationMode,
				requestId,
				message: parsed,
				account: getAddressMetaData(params.params[0], userAddressBook),
				method: params.method,
				addressBookEntries: {
					owner: getAddressMetaData(parsed.message.owner, userAddressBook),
					spender: getAddressMetaData(parsed.message.spender, userAddressBook),
					verifyingContract: getAddressMetaData(parsed.domain.verifyingContract, userAddressBook)
				},
				params,
```

File: extension/app/ts/background/windows/personalSign.ts [hunk 4]
```diff
@@ -43,6 +43,81 @@ function reject() {
...
+				requestId: requestId,
+				message: parsed,
+				account: getAddressMetaData(params.params[0], userAddressBook),
+				method: params.method,
+				addressBookEntries: {
+					token: getAddressMetaData(parsed.message.details.token, userAddressBook),
+					spender: getAddressMetaData(parsed.message.spender, userAddressBook),
+					verifyingContract: getAddressMetaData(parsed.domain.verifyingContract, userAddressBook)
+				},
+				params: params
```
Comment (MicahZoltu) [inline extension/app/ts/background/windows/personalSign.ts:102]:
```suggestion
				activeAddress,
				type: 'Permit2' as const,
				simulationMode,
				requestId,
				message: parsed,
				account: getAddressMetaData(params.params[0], userAddressBook),
				method: params.method,
				addressBookEntries: {
					token: getAddressMetaData(parsed.message.details.token, userAddressBook),
					spender: getAddressMetaData(parsed.message.spender, userAddressBook),
					verifyingContract: getAddressMetaData(parsed.domain.verifyingContract, userAddressBook)
				},
				params,
```

File: extension/app/ts/background/windows/personalSign.ts [hunk 5]
```diff
@@ -43,6 +43,81 @@ function reject() {
...
+		method: 'popup_personal_sign_request',
+		data: {
+			activeAddress,
+			type: 'EIP712' as const,
+			simulationMode: simulationMode,
+			requestId: requestId,
+			message: params.params[1],
+			account: getAddressMetaData(params.params[0], userAddressBook),
+			method: params.method,
+			params: params
```
Comment (MicahZoltu) [inline extension/app/ts/background/windows/personalSign.ts:116]:
```suggestion
			activeAddress,
			type: 'EIP712' as const,
			simulationMode,
			requestId,
			message: params.params[1],
			account: getAddressMetaData(params.params[0], userAddressBook),
			method: params.method,
			params,
```

**Review by MicahZoltu — APPROVED** [review]
Should we add an issue for fixing this in the approve access popup?

**Comment by KillariDev** [comment]
> Should we add an issue for fixing this in the approve access popup?

there's pr for it: https://github.com/DarkFlorist/TheInterceptor/pull/339

**Pull request merged**